src/main/java/net/fortuna/ical4j/transform/recurrence/AbstractDateExpansionRule.java

```diff
@@ -116,4 +117,17 @@ protected Calendar getCalendarInstance(final Date date, final boolean lenient) {
 
         return cal;
     }
+
+    /**
+     * Get a DateTime from cal.getTime() with the timezone of the given reference date.
+     *
+     * @param referenceDate
+     * @param cal
+     * @return
+     */
+    protected static Date getTime(final Date referenceDate, final Calendar cal) {
+        final Date zonedDate = new DateTime(referenceDate);
+        zonedDate.setTime(cal.getTime().getTime());
+        return zonedDate;
```

Comment:
If reference date is a DateTime object, its timezone is used for the resulting DateTime. If it's just a date object, the resulting DateTime doesn't have a TimeZone.